hexrd/wppf/WPPF.py

```diff
@@ -340,22 +340,31 @@ def calctth(self):
         self.tth = {}
         self.hkls = {}
         self.dsp = {}
+        self.sf_hkl_factors = {}
+        self.sf_lfactor = {}
         for p in self.phases:
             self.tth[p] = {}
             self.hkls[p] = {}
             self.dsp[p] = {}
+            self.sf_hkl_factors[p] = {}
+            self.sf_lfactor[p] = {}
             for k, l in self.phases.wavelength.items():
                 t = self.phases[p].getTTh(l[0].getVal('nm'))
+                sf_f, lfact_sf = self.phases[p].get_sf_hkl_factors()
                 allowed = self.phases[p].wavelength_allowed_hkls
                 t = t[allowed]
                 hkl = self.phases[p].hkls[allowed, :]
                 dsp = self.phases[p].dsp[allowed]
+                sf_f = sf_f[allowed]
                 tth_min = min(self.tth_min)
                 tth_max = max(self.tth_max)
                 limit = np.logical_and(t >= tth_min, t <= tth_max)
                 self.tth[p][k] = t[limit]
                 self.hkls[p][k] = hkl[limit, :]
                 self.dsp[p][k] = dsp[limit]
+                if sf_f is not None and lfact_sf is not None:
```

Comment:
I don't think `sf_f` could be `None`, since `sf_f = sf_f[allowed]` is above. But `sf_f` might have a `len()` of 0, or a `len()` less than or equal to the `limit` (which would cause an exception).

Reply:
Nice catch. Fixed now